src/shiv/bootstrap/__init__.py

```diff
@@ -218,7 +218,7 @@ def bootstrap():  # pragma: no cover
         if preamble_bin.suffix == ".py":
             runpy.run_path(
                 preamble_bin,
-                init_globals={"env": env, "site_packages": site_packages},
+                init_globals={"env": env, "script_path": sys.argv[0], "site_packages": site_packages},
```

Comment:
How about we call this `sys_argv` and pass the whole original `sys.argv` into it?

I can see it being helpful to know this regardless of wanting to know the script path.

Reply:
the rest of the `sys.argv` remains unchanged, so you can still reference `sys.argv` directly in preamble code. Only `sys.argv[0]` gets overwritten by [`runpy`](https://docs.docker.com/compose/compose-file/#image):

> A number of alterations are also made to the sys module. Firstly, sys.path may be altered as described above. sys.argv[0] is updated with the value of file_path and sys.modules[__name__] is updated with a temporary module object for the module being executed. All modifications to items in sys are reverted before the function returns.
>
> Note that, unlike run_module(), the alterations made to sys are not optional in this function as these adjustments are essential to allowing the execution of sys.path entries. As the thread-safety limitations still apply, use of this function in threaded code should be either serialised with the import lock or delegated to a separate process.

Reply:
My only hesitation for using "script_name" is that it could be confusing to preamble script authors... Does it refer to the preamble script name? or the actual pyz's name? Doubly so since runpy modifies `sys.argv[0]` to be the _preamble_ script name.

Reply:
@lorencarvalho I'm on board with using `archive` instead of `script_name` - it's more concise and specific. what do you think?

Reply:
@luislew I'm happy with `archive`, that's what we call it in `bootstrap/__init__.py` 👍

Reply:
Updated 👍